signer/migrations/0003__create_tables.sql

```diff
@@ -184,3 +191,111 @@ CREATE TABLE sbtc_signer.withdrawal_reject_events (
     created_at    TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
 );
 
+-- Represents a combined transaction package which is broadcasted to the Bitcoin
+-- network. The transaction package is built up of multiple transactions which
+-- are tracked separately in the `sweep_transactions` table. A transaction
+-- package may contain the sweeping transactions for both deposit and
+-- withdrawal requests.
+CREATE TABLE sbtc_signer.sweep_packages (
+    -- Internal ID of the package
+    id SERIAL PRIMARY KEY,
+    -- The Bitcoin block hash at which this package was created.
+    created_at_block_hash BYTEA NOT NULL,
+    -- The Bitcoin market fee rate at the time this package was created.
+    market_fee_rate DOUBLE PRECISION NOT NULL,
+    -- Timestamp of when this package was created.
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+-- Represents an individual transaction within a broadcasted sweep transaction
+-- package. Individual deposit and withdrawal requests reference back to these
+-- transactions to keep track of both the overall transaction package as well as
+-- the individual Bitcoin transactions they are related to.
+CREATE TABLE sbtc_signer.sweep_transactions (
+    -- Internal ID of the transaction.
+    id BIGSERIAL PRIMARY KEY,
+    -- The ID of the grouping transaction package.
+    sweep_package_id INTEGER NOT NULL,
+    -- The Bitcoin transaction ID of the transaction.
+    txid BYTEA NOT NULL,
```

Comment:
Shouldn't this be the primary key?

Reply:
It should definitely be a unique index if nothing else, good catch. I'm not a huge fan of unnecessarily using large columns as PK's in growing tables which have a lot of FK's (like in this case with swept_withdrawals and swept_deposits, 8 vs 32 bytes for each reference, and really this doesn't need to be `bigserial` either) -- unless the entire convention of the db is to use i.e. uuid surrogate keys.

Reply:
Added a unique index for `txid` in the meantime, pending the discussion on monday. [2ee116d](https://github.com/stacks-network/sbtc/pull/710/commits/2ee116d31948a4e5f2f5e95512b468aa5ecf8cb2)

Reply:
This will revert to using natural keys, i.e. txid will be the PK.

Reply:
Natural keys in [25c1b2f](https://github.com/stacks-network/sbtc/pull/710/commits/25c1b2ffd730e593228cc57e12574671aa9530cd)